packages/graph/hash_graph/lib/graph/src/store/postgres/resolve.rs

```diff
@@ -0,0 +1,164 @@
+use async_trait::async_trait;
+use error_stack::{IntoReport, Result, ResultExt};
+use tokio_postgres::{GenericClient, Row, RowStream};
+use type_system::{uri::VersionedUri, DataType, PropertyType};
+
+use crate::store::{postgres::parameter_list, AsClient, PostgresStore, QueryError};
+
+/// Context used to for [`Resolve`].
+///
+/// This is only used as an implementation detail inside of the [`postgres`] module.
+///
+/// [`Resolve`]: crate::store::query::Resolve
+/// [`postgres`]: crate::store::postgres
+// TODO: Use the context to hold query data
+//   see https://app.asana.com/0/0/1202884883200946/f
+#[async_trait]
+pub trait PostgresContext {
+    async fn read_all_data_types(&self) -> Result<RowStream, QueryError>;
+
+    async fn read_versioned_data_type(
+        &self,
+        uri: &VersionedUri,
+    ) -> Result<Record<DataType>, QueryError>;
+
+    async fn read_all_property_types(&self) -> Result<RowStream, QueryError>;
+
+    async fn read_versioned_property_type(
+        &self,
+        uri: &VersionedUri,
+    ) -> Result<Record<PropertyType>, QueryError>;
+}
+
+#[async_trait]
+impl<T> PostgresContext for &T
```

Comment:
> The blanket implementation for &T below is only for simplifying its usage for &PostgresStore, in the future I expect a custom struct for this.

Can you encapsulate this in a comment here please

Reply:
Implicity resolved by c7306f0d67b78c5d2c2850d08ffc18d0f200e70b